exchanges/bitfinex/bitfinex_types.go

```diff
@@ -395,80 +422,121 @@ type GenericResponse struct {
 
 // Position holds position information
 type Position struct {
-	ID        int64   `json:"id"`
-	Symbol    string  `json:"string"`
-	Status    string  `json:"active"`
-	Base      float64 `json:"base,string"`
-	Amount    float64 `json:"amount,string"`
-	Timestamp string  `json:"timestamp"`
-	Swap      float64 `json:"swap,string"`
-	PL        float64 `json:"pl,string"`
+	ID        int64      `json:"id"`
+	Symbol    string     `json:"string"`
+	Status    string     `json:"active"`
+	Base      float64    `json:"base,string"`
+	Amount    float64    `json:"amount,string"`
+	Timestamp types.Time `json:"timestamp"`
+	Swap      float64    `json:"swap,string"`
+	PL        float64    `json:"pl,string"`
 }
 
 // BalanceHistory holds balance history information
 type BalanceHistory struct {
-	Currency    string  `json:"currency"`
-	Amount      float64 `json:"amount,string"`
-	Balance     float64 `json:"balance,string"`
-	Description string  `json:"description"`
-	Timestamp   string  `json:"timestamp"`
+	Currency    string     `json:"currency"`
+	Amount      float64    `json:"amount,string"`
+	Balance     float64    `json:"balance,string"`
+	Description string     `json:"description"`
+	Timestamp   types.Time `json:"timestamp"`
 }
 
 // MovementHistory holds deposit and withdrawal history data
 type MovementHistory struct {
-	ID               int64   `json:"id"`
-	TxID             string  `json:"txid"`
-	Currency         string  `json:"currency"`
-	Method           string  `json:"method"`
-	Type             string  `json:"withdrawal"`
-	Amount           float64 `json:"amount,string"`
-	Description      string  `json:"description"`
-	Address          string  `json:"address"`
-	Status           string  `json:"status"`
-	Timestamp        float64 `json:"timestamp"`
-	TimestampCreated float64 `json:"timestamp_created"`
-	Fee              float64 `json:"fee"`
+	ID                 int64
+	Currency           string
+	CurrencyName       string // AKA Method
+	TXID               string
+	MTSStarted         types.Time
+	MTSUpdated         types.Time
+	Status             string
+	Amount             types.Number // Positive for deposits, negative for withdrawals
+	Fees               types.Number
+	DestinationAddress string
+	PaymentID          *string
+	TransactionID      *string
+	TransactionNote    *string
+	TransactionType    string // "deposit" or "withdrawal"
+}
+
+// UnmarshalJSON unmarshals JSON data into a MovementHistory struct
+func (m *MovementHistory) UnmarshalJSON(data []byte) error {
+	tmp := [22]any{
+		&m.ID,                 // [0]
+		&m.Currency,           // [1]
+		&m.CurrencyName,       // [2]
+		new(any),              // [3] null placeholder
```

Comment:
I think above we might be able to allocate `var placeholder json.RawMessage` and then use `placeholder` or `&placeholder` just to keep allocations down maybe? 🤷 Just a though. This is beast btw. 🔥